src/app/pages/cart/components/cart-summary/cart-summary.component.ts

```diff
@@ -36,9 +50,41 @@ export class CartSummaryComponent implements OnInit {
         this.subTotal = totalPrice;
         this.cdr.detectChanges();
       });
+
+    this.cartFacadeService.getCart()
+      .pipe(takeUntilDestroyed(this.destroyRef))
+      .subscribe((cart: ICartItem[]) => this.cart = cart);
   }
 
   buildTranslationKey(relativeKey: string): string {
     return `cart.${relativeKey}`;
   }
+
+  checkout(): void {
+    const newOrder: IOrder = {
+      products: this.cart,
+      totalAmount: this.calculateTotal,
+      status: OrderStatus.processing,
+      orderDate: this.generateOrderDate(),
+    }
+
+    this.ordersFacadeService.createOrder(newOrder);
+
+    this.cartFacadeService.clearCart();
+
+    this.router.navigate([Route.orders]);
+  }
+
+  private generateOrderDate(): string {
+    const currentDate = new Date();
+    const year = currentDate.getFullYear();
+    const month = String(currentDate.getMonth() + 1).padStart(2, '0');
+    const day = String(currentDate.getDate()).padStart(2, '0');
+    const hours = String(currentDate.getHours()).padStart(2, '0');
+    const minutes = String(currentDate.getMinutes()).padStart(2, '0');
+    const seconds = String(currentDate.getSeconds()).padStart(2, '0');
+    const formattedDateTime = `${ year }.${ month }.${ day } ${ hours }:${ minutes }:${ seconds }`;
+
+    return formattedDateTime.toString();
+  }
```

Comment:
1. If we have some logic that is not the task of the component, we shouldn't place it inside the component. For example, our component shouldn't know how this date can be calculated. In general, we place it in services.
2. Instead of using the `String` class constructor, we have the `.toString()` method.
3. I recommend saving dates in DB in the format of `number`, for example, you can use Date.now() for it. When we use it, we have a generic format that can be easily converted to smth else. And also we save the time zone, which can help us in not making mistakes when we show dates.
4. When you declare & init the `formattedDateTime` variable, it will be the string, so converting it to a string again will be extra.